src/mono/mono/mini/interp/interp-simd.c

```diff
@@ -132,29 +193,63 @@ interp_v128_op_bitwise_inequality (gpointer res, gpointer v1, gpointer v2)
 		*(gint32*)res = 1;
 }
 
-// Vector128<float>EqualsFloatingPoint
+// Vector128<float>.Equals
 static void
-interp_v128_r4_float_equality (gpointer res, gpointer v1, gpointer v2)
+interp_v128_instance_equals_r4 (gpointer res, gpointer v1, gpointer v2)
 {
-	v128_r4 v1_cast = *(v128_r4*)v1;
+	v128_r4 v1_cast = **(v128_r4**)v1;
```

Comment:
this feels wrong, shouldn't v1 and v2 both have the same level of indirection?

Reply:
It’s an instance method so v1 is technically the `this` pointer and functionally `in Vector128<T>`

Reply:
Oooh, I see. That makes sense, then. I would suggest renaming the arg from v1 to make that clearer.